services/web-app/components/svg-libraries/icon-library/icon-library.js

```diff
@@ -0,0 +1,132 @@
+import { Column, Grid } from '@carbon/react'
+import { debounce, groupBy } from 'lodash'
+import dynamic from 'next/dynamic'
+import { useEffect, useState } from 'react'
+
+import FilterRow from '@/components/svg-libraries/filter-row'
+import useColumnCount from '@/utils/use-column-count'
+
+import NoResult from '../no-result'
+import styles from '../svg-library.module.scss'
+import IconCategory from './icon-category'
+
+const { icons: iconMetaData, categories: iconCategoryMetadata } = metaData
+
+const IconLibrary = () => {
+  const [iconComponents, setIconComponents] = useState([])
+  const [selectedCategory, setSelectedCategory] = useState('All icons')
+  const [searchInputValue, setSearchInputValue] = useState('')
+  const [categoryList, setCategoryList] = useState([])
+  const [categoriesLoaded, setCategoriesLoaded] = useState(false)
+  const columnCount = useColumnCount({ assetType: 'icons' })
+
+  const debouncedSetSearchInputValue = debounce(setSearchInputValue, 200)
+
+  useEffect(() => {
+    const iconArray = iconMetaData.reduce((accumulator, icon) => {
+      if (icon.deprecated) {
+        return accumulator
+      }
+
+      const path = icon.moduleInfo.filepath
+
+      if (icon.sizes.length === 1 && icon.sizes[0] === 'glyph') {
+        return [
+          ...accumulator,
+          {
+            ...icon,
+            Component: dynamic(() => import(`@carbon/icons-react/lib/${path}`))
+          }
+        ]
+      }
+      return [
+        ...accumulator,
+        {
+          ...icon,
+          Component: dynamic(() => import(`@carbon/icons-react/lib/${path}`))
+        }
+      ]
+    }, [])
+
+    setCategoryList(
+      iconCategoryMetadata
+        .flatMap(({ subcategories }) => subcategories.flatMap(({ name }) => name))
+        .sort()
+    )
+
+    setCategoriesLoaded(true)
+
+    setIconComponents(iconArray)
+  }, [])
+
+  const getFilteredIcons = () => {
+    if (!searchInputValue) {
+      return iconComponents
+    }
+    return iconComponents.filter(({ friendlyName, category, subcategory, aliases = [], name }) => {
+      const searchValue = searchInputValue.toLowerCase()
+      return (
+        friendlyName.toLowerCase().includes(searchValue) ||
+        aliases.some((alias) => alias.toString().toLowerCase().includes(searchValue)) ||
+        subcategory.toLowerCase().includes(searchValue) ||
+        category.toLowerCase().includes(searchValue) ||
+        name.toLowerCase().includes(searchValue)
+      )
+    })
+  }
+
+  const filteredIcons = getFilteredIcons()
+
+  const allCategories = Object.entries(groupBy(filteredIcons, 'subcategory')).sort()
+
+  const filteredCategories =
+    selectedCategory === 'All icons'
+      ? allCategories
+      : allCategories.filter(([category]) => category === selectedCategory)
+
+  const shouldShowNoResult = !!(categoriesLoaded && filteredCategories.length < 1)
+
+  return (
+    <Grid condensed>
+      <Column sm={4} md={8} lg={16}>
```

Comment:
```suggestion
      <Column sm={4} md={8} lg={12}>
```

Reply:
This should fix the subgrid columns showing up the incorrect width.

Reply:
Thanks!!